### PR TITLE
ci: deploy docs after release only

### PR DESCRIPTION
**Pull request by AriPerkkio**
Deploy documentation only when release is published. 

- https://developers.cloudflare.com/pages/how-to/use-direct-upload-with-continuous-integration/#set-up-a-workflow

### CODE REVIEW

File: .github/workflows/publish-release.yaml [hunk 1]
```diff
@@ -42,6 +42,19 @@ jobs:
...
           git push origin ${{ steps.resolve-release-version.outputs.version }}
 
+      - name: Build docs
+        run: pnpm run docs:build
+
+      - name: Deploy documentation
+        uses: cloudflare/pages-action@v1
+        with:
+          apiToken: ${{ secrets.CLOUDFLARE_API_TOKEN }}
+          accountId: ${{ secrets.CLOUDFLARE_ACCOUNT_ID }}
```
Comment (AriPerkkio) [inline .github/workflows/publish-release.yaml:52]: We'll need these set up before merging

**Review by Nemikolh — APPROVED** [review]
Looks good! :+1:

File: .github/workflows/publish-release.yaml (hunk 1 above)
Comment (AriPerkkio) [reply .github/workflows/publish-release.yaml:52]: These are now set ✅

**Comment by AriPerkkio** [comment]
Automatic deployments of `tutorialkit-docs-page` are now disabled. The demo-page is still deployed automatically like before.

**Pull request merged**